src/mpid/ch4/include/mpidpre.h

```diff
@@ -96,6 +96,7 @@ typedef struct MPIDIG_rreq_t {
     MPIR_Request *match_req;
     MPIR_Request *request;
 
+    MPI_Aint data_sz;
```

Comment:
Shouldn't this field be stored in `recv_async`? Then you don't need repeat it in each of the following union struct.

Reply:
Let me think about how to get rid of this.

Reply:
@hzhou I have come up with a different solution. This `data_sz` is no longer needed as the value is already stored in the request for unexpected long send, I just need to move the init of recv before sending the cts.

As for RMA related ones, most of them do not need the data_sz being sent as they can be calculated from the datatype signature (which we already do in some places). I ended up just clean out it a bit.

Is this looks better now?

Reply:
Yes!